### PR TITLE
[somfytahoma] improved silent mode control

### PR DESCRIPTION
**Pull request by octa22**
This PR improves silent mode control of the Silent Roller Shutter thing.
Now it moves the roller shutter silently even if the UP or DOWN command received.

Confirmed to be working
https://community.openhab.org/t/re-tahoma-binding-compatible-with-oh2/100430/10

### CODE REVIEW

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java [hunk 1]
```diff
@@ -45,25 +45,37 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                    }
+                    break;
+                default:
+                    if (CONTROL_SILENT.equals(channelUID.getId()) && (COMMAND_UP.equals(cmd) || COMMAND_DOWN.equals(cmd))) {
+                        // move the roller shutter to the specific position at low speed
+                        String param = "[" + (COMMAND_UP.equals(cmd) ? 0 : 100) + ", \"lowspeed\"]";
+                        sendCommand(COMMAND_SET_CLOSURESPEED, param);
+                    } else {
+                        sendCommand(cmd);
+                    }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java:78]:
```suggestion
                case COMMAND_UP:
                case COMMAND_DOWN:
                    if (CONTROL_SILENT.equals(channelUID.getId())) {
                        // move the roller shutter to the specific position at low speed
                        String param = "[" + (COMMAND_UP.equals(cmd) ? 0 : 100) + ", \"lowspeed\"]";
                        sendCommand(COMMAND_SET_CLOSURESPEED, param);
                        break;
                    }
                    //fall through
                default:
                    sendCommand(cmd);
```

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java [hunk 2]
```diff
@@ -45,25 +45,37 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                    break;
+                case COMMAND_STOP:
+                    String executionId = getCurrentExecutions();
+                    if (executionId != null) {
+                        // Check if the roller shutter is moving and STOP is sent => STOP it
+                        cancelExecution(executionId);
+                    } else {
+                        sendCommand(COMMAND_MY);
+                    }
+                    break;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java:60]: You can do similar fall through shenanigans here as well by moving the COMMAND_MY case around.
```suggestion
                case COMMAND_STOP:
                    String executionId = getCurrentExecutions();
                    if (executionId != null) {
                        // Check if the roller shutter is moving and STOP is sent => STOP it
                        cancelExecution(executionId);
                        break;
                    }
                    //fall through
                case COMMAND_MY:
                    sendCommand(COMMAND_MY);
                    break;
```

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java (hunk 1 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java:78]: Great suggestion. thanks

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java (hunk 2 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaSilentRollerShutterHandler.java:60]: Great suggestion. thanks

**Pull request merged**